### PR TITLE
Replace misleading documentation about action inputs and outputs

### PR DESCRIPTION
**Pull request by piotrekkr**
### Why:

Current docs for `inputs` and `outputs` in GitHub action metadata syntax contains some misleading statements

> ## `inputs`
> 
> **Optional** Input parameters allow you to specify data that the action expects to use during runtime. {% data variables.product.prodname_dotcom %} stores input parameters as environment variables. Input ids with uppercase letters are converted to lowercase during runtime. We recommend using lowercase input ids.

I've tested this `Input ids with uppercase letters are converted to lowercase during runtime.` using 
```yaml
name: Demo Composite Action
description: Demo Composite Action

inputs:
  first-input ID:
    description: First input
    required: true
  second-INPUT_id:
    description: Second input
    required: false
    default: 1
  third_input_id:
    description: Third input
    required: false
runs:
  using: composite
  steps:
    - shell: bash
      run: env | grep ^INPUT_ || true
    - shell: bash
      run: echo '${{ toJSON(inputs) }}'

```
and nothing is converted to lowercase

image

> Note: Workflows using required: true will not automatically return an error if the input is not specified for events that automatically trigger workflow runs. If you set required: true in your workflow file and are using workflow_dispatch to manually run the workflow, you will be required to specify inputs on GitHub. For more information, see "[Events that trigger workflows](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows)."
> 
> Workflow files that use this action can use the with keyword to set an input value for octocat-eye-color. For more information about the with syntax, see "[Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepswith)."

This is misleading because documentation is about action metadata syntax and note is describing inputs in workflow metadata syntax. This has nothing to do with action inputs, I think.

> When you specify an input in a workflow file or use a default input value, {% data variables.product.prodname_dotcom %} creates an environment variable for the input with the name `INPUT_<VARIABLE_NAME>`. The environment variable created converts input names to uppercase letters and replaces spaces with `_` characters.

Again, this paragraph seems to be mixing workflow inputs with action inputs.

> If the action is written using a [composite](/actions/creating-actions/creating-a-composite-action), then it will not automatically get `INPUT_<VARIABLE_NAME>`. If the conversion doesn't occur, you can change these inputs manually.

Last sentence does not make much sense and I tested it. There is no `INPUT_*` variables present at all with composite actions. I don't even know what `you can change these inputs manually` part actually means. Inputs are read-only afaik, right?

> ### `inputs.<input_id>`
> 
> **Required** A `string` identifier to associate with the input. The value of `<input_id>` is a map of the input's metadata. The `<input_id>` must be a unique identifier within the `inputs` object. The `<input_id>` must start with a letter or `_` and contain only alphanumeric characters, `-`, or `_`.

As far as my tests got me, `space` characters are also valid to use in input ids.

> ### `inputs.<input_id>.description`
>
> **Required** A `string` description of the input parameter.

I tested this with composite and docker action:

```yaml
name: Demo Docker Action
description: Demo Docker Action

inputs:
  first-input ID:
    required: true
  second-INPUT_id:
    required: false
    default: 1
  third_input_id:
    required: false

runs:
  using: 'docker'
  image: 'Dockerfile'
  args:
    - ${{ inputs['first-input ID'] }}
    - ${{ inputs.second-INPUT_id }}
    - ${{ inputs.third_input_id }}
```
and 
```yaml
name: Demo Composite Action
description: Demo Composite Action

inputs:
  first-input ID:
    required: true
  second-INPUT_id:
    required: false
    default: 1
  third_input_id:
    required: false
runs:
  using: composite
  steps:
    - shell: bash
      run: echo '${{ toJSON(inputs) }}'
```
And inputs in those action did not require any description and actions were run without problem
```yaml
jobs:
  actions-examples:
    name: Actions Reuse
    runs-on: ubuntu-latest
    steps:
      - name: Use organization composite action
        uses: pkroczynski/action-demo-composite@inputs
        with:
          first-input ID: hello
      - name: Use organization docker action
        uses: pkroczynski/action-demo-docker@main
        with:
          first-input ID: hello
```

image

> ### `outputs.<output_id>`
>
> **Required** A `string` identifier to associate with the output. The value of `<output_id>` is a map of the output's metadata. The `<output_id>` must be a unique identifier within the `outputs` object. The `<output_id>` must start with a letter or `_` and contain only alphanumeric characters, `-`, or `_`.

There seems to be also space allowed in output id. Tested with js action using `core.setOutput('output 1', "test")` and it was returned as action output.

> ### `outputs.<output_id>.description`
>
> **Required** A `string` description of the output parameter.

Tried and description can be omitted like below without any problems so it seems that it is optional.

```yaml
outputs:
  output 1: {}
```

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Updated docs a bit to reflect what is actually happening with inputs and outputs in actions.

### Check off the following:

- [x] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline (this link will be available after opening the PR).

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

### CODE REVIEW

**Comment by Lordthevopz** [comment]
dg

ในวันที่ พ. 10 ก.ค. 2024 14:13 Piotr ***@***.***> เขียนว่า:

**Comment by nguyenalex836** [comment]
@piotrekkr Thanks so much for opening a PR! I'll get this triaged for review ✨

**Review by blacktatty-multi — COMMENTED** [review]
#Sign√

File: content/actions/creating-actions/metadata-syntax-for-github-actions.md [hunk 1]
```diff
@@ -36,18 +36,12 @@ Action metadata files use YAML syntax. If you're new to YAML, you can read "[Lea
 
 ## `inputs`
 
-**Optional** Input parameters allow you to specify data that the action expects to use during runtime. {% data variables.product.prodname_dotcom %} stores input parameters as environment variables. Input ids with uppercase letters are converted to lowercase during runtime. We recommend using lowercase input ids.
+**Optional** Input parameters allow you to specify data that the action expects to use during runtime. {% data variables.product.prodname_dotcom %} stores input parameters as environment variables. Input values are always converted to strings. We recommend using lowercase input ids.
```
Comment (ericsciple) [inline content/actions/creating-actions/metadata-syntax-for-github-actions.md:39]: > Input values are always converted to strings.

I would leave this out. Environment variables are strings.

File: content/actions/creating-actions/metadata-syntax-for-github-actions.md [hunk 2]
```diff
@@ -36,18 +36,12 @@ Action metadata files use YAML syntax. If you're new to YAML, you can read "[Lea
...
 
 ### Example: Specifying inputs
 
 This example configures two inputs: `num-octocats` and `octocat-eye-color`. The `num-octocats` input is not required and will default to a value of `1`. `octocat-eye-color` is required and has no default value.
 
-{% note %}
-
-**Note:** Workflows using `required: true` will not automatically return an error if the input is not specified for events that automatically trigger workflow runs. If you set `required: true` in your workflow file and are using `workflow_dispatch` to manually run the workflow, you will be required to specify inputs on {% data variables.product.prodname_dotcom %}. For more information, see "[AUTOTITLE](/actions/using-workflows/events-that-trigger-workflows)."
```
Comment (ericsciple) [inline content/actions/creating-actions/metadata-syntax-for-github-actions.md:47]: Maybe just:

```
**Note:** Workflows using `required: true` will not automatically return an error if the input is not specified.
```

Comment (ericsciple) [inline content/actions/creating-actions/metadata-syntax-for-github-actions.md:47]: Maybe just

```
**Note:** Actions using `required: true` will not automatically return an error if the input is not specified.
```

File: content/actions/creating-actions/metadata-syntax-for-github-actions.md [hunk 3]
```diff
@@ -61,29 +55,29 @@ inputs:
...
 
 For example, if a workflow defined the `num-octocats` and `octocat-eye-color` inputs, the action code could read the values of the inputs using the `INPUT_NUM-OCTOCATS` and `INPUT_OCTOCAT-EYE-COLOR` environment variables.
 
 ### `inputs.<input_id>`
 
-**Required** A `string` identifier to associate with the input. The value of `<input_id>` is a map of the input's metadata. The `<input_id>` must be a unique identifier within the `inputs` object. The `<input_id>` must start with a letter or `_` and contain only alphanumeric characters, `-`, or `_`.
+**Required** A `string` identifier to associate with the input. The value of `<input_id>` is a map of the input's metadata. The `<input_id>` must be a unique identifier within the `inputs` object. The `<input_id>` must start with a letter or `_` and contain only alphanumeric characters, spaces, `-`, or `_`.
```
Comment (ericsciple) [inline content/actions/creating-actions/metadata-syntax-for-github-actions.md:68]: Spaces should never have been allowed and aren't allowed for job IDs. I do not want to advertise/encourage this.

File: content/actions/creating-actions/metadata-syntax-for-github-actions.md [hunk 4]
```diff
@@ -61,29 +55,29 @@ inputs:
...
 
 ### `inputs.<input_id>.description`
 
-**Required** A `string` description of the input parameter.
```
Comment (ericsciple) [inline content/actions/creating-actions/metadata-syntax-for-github-actions.md:78]: Let's leave this `Required`. We want to encourage descriptions, which will help with editing experiences.

File: content/actions/creating-actions/metadata-syntax-for-github-actions.md [hunk 5]
```diff
@@ -61,29 +55,29 @@ inputs:
...
+**Optional** A `string` description of the input parameter.
 
 ### `inputs.<input_id>.required`
 
 **Optional** A `boolean` to indicate whether the action requires the input parameter. Set to `true` when the parameter is required.
 
 ### `inputs.<input_id>.default`
 
-**Optional** A `string` representing the default value. The default value is used when an input parameter isn't specified in a workflow file.
+**Optional** A `string` representing the default value. The default value is used when an input parameter isn't specified in a workflow file. If input is not required and default value is not specified, {% data variables.product.prodname_dotcom %} will use empty string as a default value.
```
Comment (ericsciple) [inline content/actions/creating-actions/metadata-syntax-for-github-actions.md:80]: I would suggest leaving the paragraph as is (simpler), unless there is a strong reason. Would prefer to keep the docs simpler.

File: content/actions/creating-actions/metadata-syntax-for-github-actions.md [hunk 6]
```diff
@@ -107,11 +101,11 @@ outputs:
 
 ### `outputs.<output_id>`
 
-**Required** A `string` identifier to associate with the output. The value of `<output_id>` is a map of the output's metadata. The `<output_id>` must be a unique identifier within the `outputs` object. The `<output_id>` must start with a letter or `_` and contain only alphanumeric characters, `-`, or `_`.
+**Required** A `string` identifier to associate with the output. The value of `<output_id>` is a map of the output's metadata. The `<output_id>` must be a unique identifier within the `outputs` object. The `<output_id>` must start with a letter or `_` and contain only alphanumeric characters, space, `-`, or `_`.
```
Comment (ericsciple) [inline content/actions/creating-actions/metadata-syntax-for-github-actions.md:104]: same as above

File: content/actions/creating-actions/metadata-syntax-for-github-actions.md [hunk 7]
```diff
@@ -107,11 +101,11 @@ outputs:
...
 
 ### `outputs.<output_id>.description`
 
-**Required** A `string` description of the output parameter.
+**Optional** A `string` description of the output parameter.
```
Comment (ericsciple) [inline content/actions/creating-actions/metadata-syntax-for-github-actions.md:108]: same as above

**Comment by nguyenalex836** [comment]
@ericsciple Thank you very much for the review! 💛 

@piotrekkr Once you've had a chance to update your PR per the suggestions @ericsciple made, we'll be happy to get this merged ✨

**Comment by piotrekkr** [comment]
Hello again. I was on vacations for a week and was unable to respond to this and seems it was automatically closed. @nguyenalex836 Should I create new pr or this one can be reopened somehow? Thanks

**Comment by nguyenalex836** [comment]
@piotrekkr Apologies for the overzealous automation! I've just reopened this issue 💛

File: content/actions/creating-actions/metadata-syntax-for-github-actions.md (hunk 2 above)
Comment (piotrekkr) [reply content/actions/creating-actions/metadata-syntax-for-github-actions.md:47]: Done

File: content/actions/creating-actions/metadata-syntax-for-github-actions.md (hunk 1 above)
Comment (piotrekkr) [reply content/actions/creating-actions/metadata-syntax-for-github-actions.md]: Done

File: content/actions/creating-actions/metadata-syntax-for-github-actions.md (hunk 3 above)
Comment (piotrekkr) [reply content/actions/creating-actions/metadata-syntax-for-github-actions.md]: Done

File: content/actions/creating-actions/metadata-syntax-for-github-actions.md (hunk 4 above)
Comment (piotrekkr) [reply content/actions/creating-actions/metadata-syntax-for-github-actions.md]: Done

File: content/actions/creating-actions/metadata-syntax-for-github-actions.md (hunk 5 above)
Comment (piotrekkr) [reply content/actions/creating-actions/metadata-syntax-for-github-actions.md]: Done

File: content/actions/creating-actions/metadata-syntax-for-github-actions.md (hunk 6 above)
Comment (piotrekkr) [reply content/actions/creating-actions/metadata-syntax-for-github-actions.md]: Done

File: content/actions/creating-actions/metadata-syntax-for-github-actions.md (hunk 7 above)
Comment (piotrekkr) [reply content/actions/creating-actions/metadata-syntax-for-github-actions.md]: Done

**Comment by piotrekkr** [comment]
@ericsciple Unless I've missed something, all is done. Let me know if anything else should be changed. I don't have good internet connection atm and cannot clone full repo to resolve conflicts. Will do this next week.

**Review by nguyenalex836 — DISMISSED** [review]

**Comment by nguyenalex836** [comment]
@piotrekkr Thank you for addressing that feedback! I've approved the PR - once you are able to resolve those conflicts, we can get this merged 💛

**Comment by piotrekkr** [comment]
@nguyenalex836 I've rebased this PR on latest code from main.

**Review by nguyenalex836 — APPROVED** [review]

**Comment by nguyenalex836** [comment]
@piotrekkr Thank you! Getting this merged now 💛

**Pull request merged**